bottom/unpack.py

```diff
@@ -300,6 +300,21 @@ def unpack_command(msg: str) -> Tuple[str, Dict[str, Any]]:
         else:
             kwargs["message"] = ""
 
+    elif command in ["MODE"]:
+        nickmask(prefix, kwargs)
+        if "#" in params[0]:
```

Comment:
This is simply a convention, right?  Do we have any stronger way to tell apart user/channel mode commands?  For example, minimum arg lengths...

Here are the examples from [rfc 2812](https://www.rfc-editor.org/rfc/rfc2812.txt).

3.1.5 User mode message

```
(1)MODE WiZ -w                     ; Command by WiZ to turn off
                                   reception of WALLOPS messages.

(2)MODE Angel +i                   ; Command from Angel to make herself
                                   invisible.

(3)MODE WiZ -o                     ; WiZ 'deopping' (removing operator
                                   status).
```

3.2.3 Channel mode message

```
(1)MODE #Finnish +imI *!*@*.fi     ; Command to make #Finnish channel
                                   moderated and 'invite-only' with user
                                   with a hostname matching *.fi
                                   automatically invited.

(2)MODE #Finnish +o Kilroy         ; Command to give 'chanop' privileges
                                   to Kilroy on channel #Finnish.

(3)MODE #Finnish +v Wiz            ; Command to allow WiZ to speak on
                                   #Finnish.

(4)MODE #Fins -s                   ; Command to remove 'secret' flag
                                   from channel #Fins.

(5)MODE #42 +k oulu                ; Command to set the channel key to
                                   "oulu".

(6)MODE #42 -k oulu                ; Command to remove the "oulu"
                                   channel key on channel "#42".
```


I think there are completely ambiguous cases eg. 3.1.5 (1) and 3.2.3 (4) are both `MODE [^\s]+ [^\s]+`, in which case we either:
1. Codify the convention that any first parameter starting with the literal "#" is a channel command; user otherwise.
2. Leave it up to the user to parse, and only expose one command `@client.on("MODE")`.

For the first I decided to see if channels or nicks have specific constraints.  There's this bit, from "1.3 Channels":

> 1.3 Channels
> 
>    Channels names are strings (beginning with a '&', '#', '+' or '!'
>    character) of length up to fifty (50) characters.  Apart from the
>    requirement that the first character is either '&', '#', '+' or '!',
>    the only restriction on a channel name is that it SHALL NOT contain
>    any spaces (' '), a control G (^G or ASCII 7), a comma (',').  Space
>    is used as parameter separator and command is used as a list item
>    separator by the protocol).  A colon (':') can also be used as a
>    delimiter for the channel mask.  Channel names are case insensitive.

So I'm ok with the first option (splitting the command into two modes based on `params[0]`) as long as we check that `params[0] in "&#+!"`.  If that's too restrictive, I'm +1 for the second option of simply exposing `MODE` as a single event that unpacks to a dict with keys `[*nickmask, target, modes, params]` where target is either a nickname or channel.

Reply:
I only split the MODE to keep it consistent with the library return. We can keep it the same way, as IRC clients do, and return MODE. Imho I think it's a good idea to keep it consistent in a way that pack and unpack work with the same way. We can send MODE and trigger MODE or do what we did and split them into 2. There are a lot of ambiguous cases in the RFC where it keeps things unclear.